sqlglot/generator.py

```diff
@@ -1255,6 +1268,20 @@ def withdataproperty_sql(self, expression: exp.WithDataProperty) -> str:
             statistics_sql = f" AND {'NO ' if not statistics else ''}STATISTICS"
         return f"{data_sql}{statistics_sql}"
 
+    def withsystemversioningproperty_sql(self, expression: exp.WithSystemVersioningProperty) -> str:
+        sql = f"WITH(SYSTEM_VERSIONING=ON"
+        if expression.this is None:
+            return f"{sql})"
```

Comment:
This branch is inconsistent with the fact that `this` is mandatory in the definition of `WithSystemVersioningProperty`. You need to either have `this` be optional (i.e. map to `False`) or rewrite this section because it seems like dead code right now.

Reply:
It seems like it's not tested too - let's make sure we test all of the new execution paths